# Embeddings - AutoTokenizer

This notebook includes experimentation with the Embeddings through the usage of the AutoTokenizer.

# Setup Notebook

## Imports

In [5]:
# Import Standard Libraries
import json
from transformers import AutoTokenizer, AutoModel
import torch

# Experimentations

## BERT

### Tokenization

In [3]:
# Instance the Tokenizer
tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")

tokenizer_config.json:   0%|          | 0.00/48.0 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/570 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/232k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/466k [00:00<?, ?B/s]

- The Tokenizer is used to converts raw text into tokens
<br>

**The Process:**
1) The first step is WordPiece Tokenization (e.g., "playing" &rarr; ["play", "##ing"])
2) The second step maps tokens into numerical IDs (based on BERT's vocabulary)
3) Add special tokens (`[CLS]` and `[SEP]`)
4) Pads and truncate text sequence to fix model's input
5) Create attention mask

In [18]:
text = "I love Data Science"
tokens = tokenizer(text, return_tensors="pt")  # Convert to PyTorch tensors

print('Tokens:' , tokens.input_ids)
print('Attention Mask:', tokens.attention_mask)

Tokens: tensor([[ 101, 1045, 2293, 2951, 2671,  102]])
Attention Mask: tensor([[1, 1, 1, 1, 1, 1]])


- [101] = [CLS] token (start of sentence)
- 1045 = "I", 2293 = "love", 2951 = "data", 2671 = "science"
- [102] = [SEP] token (end of sentence)

### Embeddings